artichoke-backend/src/extn/core/array/array.rb

```diff
@@ -1105,12 +1105,20 @@ def select!(&block)
     replace(result)
   end
 
-  def shuffle(_rng = (not_set = true)) # rubocop:disable Lint/UselessAssignment
-    raise NotImplementedError, 'TODO implement in Rust'
+  def shuffle(rng = nil)
```

Comment:
the default assignment initializer is required because passing `nil` to `Array#shuffle` should raise an exception.

When `not_set` is `true`, `Array#shuffle` should use `Random::Default`